.teamcity/Gradle_Check/projects/FunctionalTestProject.kt

```diff
@@ -21,7 +22,16 @@ class FunctionalTestProject(model: CIBuildModel, testConfig: TestCoverage, stage
             return@forEach
         }
 
-        buildType(FunctionalTest(model, testConfig, bucket.getSubprojectNames(), stage, bucket.name, bucket.extraParameters()))
+        if (bucket is SubprojectSplit) {
+            // https://github.com/gradle/gradle-private/issues/2693
+            if (testConfig.testType.supportTestSplit) {
```

Comment:
I guess we could move this logic to the `SubprojectSplit` by adding a (or a few) methods to the `BuildTypeBucket` interface.